app/webpacker/components/CompetitionsOverview/filterUtils.js

```diff
@@ -1,4 +1,19 @@
-import { events } from '../../lib/wca-data.js.erb';
+import { DateTime } from 'luxon';
+import {
+  events, continents, countries,
+} from '../../lib/wca-data.js.erb';
+
+// constants
+
+const WCA_EVENT_IDS = Object.keys(events.byId);
+
+const YEARS_WITH_PAST_COMPETITIONS = [];
+for (let { year } = DateTime.now(); year >= 2003; year -= 1) {
+  YEARS_WITH_PAST_COMPETITIONS.push(year);
+}
+YEARS_WITH_PAST_COMPETITIONS.push(1982);
+
+export { YEARS_WITH_PAST_COMPETITIONS };
```

Comment:
I don't think @gregorbg will accept the generation of years list this way. This should have been missed by Gregor in the review of https://github.com/thewca/worldcubeassociation.org/pull/7902.

I think what Gregor suggests will be to export this in `wca-data.js.erb` in the following way:
```
export const YEARS_WITH_PAST_COMPETITIONS = <%= Competition.all.map(&:start_date).map(&:year).uniq.sort %>
```

Reply:
Hehe, I've been reviewing too much of your code recently, it seems :stuck_out_tongue: 

As a matter of fact, I wouldn't call it "missed". More like "be aware that we have developers with different degrees of experience and different aspirations" :wink: 

But indeed, your suggestion looks more concise. Except for the fact that `Competition.all.map(...)` loads all competitions from our DB into memory via a `SELECT * FROM Competitions` call and then loops over them only for retrieving the start date. That's a complete waste of resources, and can be improved with `Competition.pluck(:start_date)` which translates to a `SELECT start_date FROM Competitions` SQL call